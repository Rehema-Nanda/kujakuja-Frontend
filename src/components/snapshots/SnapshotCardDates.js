import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const SnapshotCardDates = (props) => {
    const { singleDay, start, end } = props;
    const sameMonth = start.month() === end.month();
    if (singleDay) {
        return <p className="previous-next">{start.format('ddd D')}</p>;
    } else {
        return (
            <p className="previous-next">
                {start.format(`ddd D${!sameMonth ? ' MMM' : ''}`)}
            <br />
            ‚ûù
                &nbsp;
                {end.format(`ddd D${!sameMonth ? ' MMM' : ''}`)}
            </p>
        );
    }
};

SnapshotCardDates.propTypes = {
    start: PropTypes.instanceOf(moment),
    end: PropTypes.instanceOf(moment),
    singleDay: PropTypes.bool,
};

export default SnapshotCardDates;
