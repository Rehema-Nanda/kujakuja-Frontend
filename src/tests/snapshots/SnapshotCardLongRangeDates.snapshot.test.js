import React from 'react';
import renderer from 'react-test-renderer';
import SnapshotCardDates from '../../components/snapshots/SnapshotCardDates';
import moment from 'moment';

it('renders correctly', () => {
    let fixedDate = moment.utc('2019-07-17').startOf('day');

    const tree = renderer.create(
        <SnapshotCardDates
            singleDay={false}
            start={fixedDate.clone().subtract(32, 'days')}
            end={fixedDate}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
