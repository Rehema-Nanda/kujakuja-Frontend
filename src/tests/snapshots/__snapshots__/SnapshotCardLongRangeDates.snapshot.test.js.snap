// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="previous-next"
>
  Sat 15 Jun
  <br />
  ➝  
  Wed 17 Jul
</p>
`;
